patches/pickling.patch

```diff
@@ -0,0 +1,35 @@
+Only in esutil-0.6.2.5: build
+diff -ru esutil-0.6.2.5-orig/esutil/htm/htm.py esutil-0.6.2.5/esutil/htm/htm.py
+--- esutil-0.6.2.5-orig/esutil/htm/htm.py	2017-09-25 08:28:03.000000000 -0700
++++ esutil-0.6.2.5/esutil/htm/htm.py	2019-05-09 12:52:34.670264752 -0700
+@@ -689,6 +689,11 @@
+         else:
+             return counts
+ 
++    def __reduce__(self):
++        """To support pickle/unpickle: only depth matters"""
++        return (HTM, (self.get_depth(), ))
++
++
+ class Matcher(htmc.Matcher):
+     """
+     Object to match arrays of ra,dec
+diff -ru esutil-0.6.2.5-orig/esutil/unit_tests/htm_tests.py esutil-0.6.2.5/esutil/unit_tests/htm_tests.py
+--- esutil-0.6.2.5-orig/esutil/unit_tests/htm_tests.py	2017-09-25 08:28:03.000000000 -0700
++++ esutil-0.6.2.5/esutil/unit_tests/htm_tests.py	2019-05-09 12:51:17.816882703 -0700
+@@ -1,5 +1,6 @@
+ from __future__ import with_statement, print_function
+ import sys, os
++import pickle
+ import tempfile
+ import warnings
+ import numpy
+@@ -197,3 +198,8 @@
+                         "expected %s counts in bincount "
+                         "with array scale, got %s" % (counts_truth,counts))
+ 
++    def test_pickle(self):
++        h = htm.HTM(5)
++        pickled = pickle.dumps(h, protocol=pickle.HIGHEST_PROTOCOL)
```

Comment:
Why `HIGHEST_PROTOCOL`? I'm guessing (without having looked at the meas_algorithms code) that this would usually be called without a `protocol` argument, so it's probably most useful to test `DEFAULT_PROTOCOL` instead (or even `range(DEFAULT_PROTOCOL+1)` or something).

Reply:
`HIGHEST_PROTOCOL` results in much smaller and faster pickling, and is purely a `pickle`-internals thing; the pickle infrastructure here doesn't know or care about it. Though, in this case it shouldn't matter, since the only thing being pickled is an `int` and a function pointer. `meas_algorithms` doesn't do any pickling itself, `multiprocessing` does, and it would almost certainly use the best protocol for that.

Reply:
Multiprocessing uses the default protocol: https://github.com/python/cpython/blob/master/Lib/multiprocessing/reduction.py#L49

Just seems weird to force the test to use something else without explanation.

Reply:
Huh. I'm quite surprised they use the default there: I would have expected HIGHEST, because it's purely in-memory. Ah well, at least DEFAULT is now 3, which is good enough. I've removed the explicit protocol.